src/jupyter_contrib_nbextensions/nbextensions/hide_input/main.js

```diff
@@ -3,32 +3,85 @@
 define([
     'jquery',
     'base/js/namespace',
-    'base/js/events'
+    'base/js/events',
+    'base/js/utils',
+    'services/config'
 ], function(
     $,
     Jupyter,
-    events
+    events,
+    utils,
+    configmod
 ) {
     "use strict";
 
     var toggle_selected_input = function () {
         // Find the selected cell
         var cell = Jupyter.notebook.get_selected_cell();
         // Toggle visibility of the input div
-        cell.element.find("div.input").toggle('slow');
-        cell.metadata.hide_input = ! cell.metadata.hide_input;
+        if(Jupyter.notebook.metadata.hide_cellprompt){
+                cell.element.find("div.input").toggle('slow')
+        }else{
+            cell.element.find("div.input > div.inner_cell > div.input_area").toggle('slow')
+            cell.metadata.hide_input = ! cell.metadata.hide_input;
```

Comment:
I think this should be done regardless of the value of `Jupyter.notebook.metadata.hide_cellprompt`, no?

Reply:
The div.input contains the header and with that the celltoolbar. So line 23 should be only performed if _hide_cellprompt_ is true. What I should probably change is to force to show _div.input_ when _hide_cellprompt_ is false and to show _div.input_prompt_ as well as _div_input_area_